### PR TITLE
Java Checkstyle checks

### PR DESCRIPTION
**Pull request by aberke220**
Add documentation and configuration for checkstyle usage.
Add pre-push hook to run checkstyle validation on data-access-service.
Solves [AB#2274](https://deloittedigitalat.visualstudio.com/38398c5c-11b2-4e64-98b5-5af5e65d6e82/_workitems/edit/2274).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,3 +15,32 @@ The following services are part of the data access services (`data-access-servic
...
+
+# Development
+
+This section describes how to get started with development for this repository.
+
+## Required software
+
+- Git
+- JDK 11
+- TBD
```
Comment (dominicboeck) [inline README.md:27]: What exactly is to be defined? Or just a typo?

File: README.md [hunk 2]
```diff
@@ -15,3 +15,32 @@ The following services are part of the data access services (`data-access-servic
...
+- SSH key: `git clone git@github.com:DeloitteDigitalAT/core-banking.git`
+- github user/pwd (deprecated):
+`git clone https://github.com/DeloitteDigitalAT/core-banking.git`
+
+
+## Configure static code analysis tools
+- Checksyle: used to enforce Java coding convenions and formatting specified in `/config/checkstyle/checkstyle.xml`.
+Configuration guide is available at https://github.com/DeloitteDigitalAT/core-banking/blob/master/config/checkstyle/README.md
+- Sonar
+- TBD
```
Comment (dominicboeck) [inline README.md:45]: Same as above

File: data-access-services/pom.xml [hunk 3]
```diff
@@ -49,6 +49,15 @@
                         </excludes>
                     </configuration>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-checkstyle-plugin</artifactId>
+                    <version>3.1.2</version>
+                    <configuration>
+                        <configLocation>../config/checkstyle/checkstyle.xml</configLocation>
+                        <violationSeverity>warning</violationSeverity>
```
Comment (dominicboeck) [inline data-access-services/pom.xml:58]: Should we check test sources too?

`<includeTestSourceDirectory>true</includeTestSourceDirectory>`

File: data-access-services/pom.xml [hunk 4]
```diff
@@ -49,6 +49,15 @@
                         </excludes>
                     </configuration>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-checkstyle-plugin</artifactId>
+                    <version>3.1.2</version>
```
Comment (dominicboeck) [inline data-access-services/pom.xml:55]: Maybe declare the version in the properties above, so they can be viewed in one place at the top of the file?

File: README.md (hunk 1 above)
Comment (aberke220) [reply README.md:27]: to add other required sw (e.g. for JS development, Docker) later

File: README.md (hunk 2 above)
Comment (aberke220) [reply README.md:45]: some details should be added here about Sonar + JS lint later

File: data-access-services/pom.xml (hunk 3 above)
Comment (aberke220) [reply data-access-services/pom.xml:58]: sure, updated.

File: data-access-services/pom.xml (hunk 4 above)
Comment (aberke220) [reply data-access-services/pom.xml:55]: sure, updated.

**Review by theqajedi — APPROVED** [review]

**Pull request merged**